### PR TITLE
refactor: overwrite ArbitrageMinGasPrice config from .005 to 0.1

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

To address the failing arbitrage spam taking up block space, we will increase [arbitrage-min-gas-fee](https://github.com/osmosis-labs/osmosis/blob/3aaf5d117f96df0092b0375d0c8de0a12850d092/cmd/osmosisd/cmd/root.go#L540) to be near the order of magnitude of 0.1 OSMO.

CL Swap is around 600K gas wanted and 340K gas used. Let's assume a middle ground of 450K and aim to have the arb pay approximately 0.05 OSMO.
```
0.045 = 450_000 * 0.1 / 10**6
```
 As a result, 0.1 is an appropriate choice.

## Testing and Verifying

Tested the following vectors:
- new config
- does no overwrite other values
- old value is "0.005" -> overwritten
- old value is ".005" -> overwritten
- old value is something else -> does not change

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: cmd/osmosisd/cmd/root.go [hunk 1]
```diff
@@ -477,6 +482,77 @@ func overwriteConfigTomlValues(serverCtx *server.Context) error {
...
+
+				// Find the opening and closing quotes
+				openQuoteIndex := strings.Index(fileContent[index:], "\"")
+				openQuoteIndex += index
+
+				closingQuoteIndex := strings.Index(fileContent[openQuoteIndex+1:], "\"")
+				closingQuoteIndex += openQuoteIndex + 1
+
+				// Replace the old value with the new value
+				newFileContent := fileContent[:openQuoteIndex+1] + newArbitrageMinGasFeeValue + fileContent[closingQuoteIndex:]
```
Comment (p0mvn) [inline cmd/osmosisd/cmd/root.go:543]: Note:

The following are the reasons why this is done manually:
- https://github.com/pelletier/go-toml does not handle comments when writing. So all config comments would be removed if I were to load the file, update the value, and overwrite.
- SDK helpers either unexpectedly overwrite other config values or run into the same comment issue as above. I found it easier to simply string-match

Comment (p0mvn) [reply cmd/osmosisd/cmd/root.go:543]: https://github.com/cosmos/cosmos-sdk/issues/19257

**Review by PaddyMc — APPROVED** [review]

File: cmd/osmosisd/cmd/root.go (hunk 1 above)
Comment (czarcas7ic) [reply cmd/osmosisd/cmd/root.go:543]: I also ran into this when I attempted this a while ago, thanks for making an issue

**Review by czarcas7ic — APPROVED** [review]
Nice work

File: cmd/osmosisd/cmd/root.go [hunk 2]
```diff
@@ -477,6 +482,77 @@ func overwriteConfigTomlValues(serverCtx *server.Context) error {
...
+			serverCtx.Viper.Set(mempoolConfigName+"."+arbitrageMinGasFeeConfigName, newArbitrageMinGasFeeValue)
+
+			defer func() {
+				if err := recover(); err != nil {
+					fmt.Printf("failed to write to %s: %s\n", configFilePath, err)
+				}
+			}()
+
+			// Check if the file is writable
+			if fileInfo.Mode()&os.FileMode(0200) != 0 {
```
Comment (czarcas7ic) [inline cmd/osmosisd/cmd/root.go:519]: I think this implies if the file is not writeable, then the defer func wont trigger so there will be no feedback for the operator right?

Comment (p0mvn) [reply cmd/osmosisd/cmd/root.go:519]: They would just not have their config updated which IMO is fine. The intention is to silently ignore the update if their file isn't writeable

Comment (czarcas7ic) [reply cmd/osmosisd/cmd/root.go:519]: Right, I just think they would want to still get a log print saying it wasn't overwritten due to  permission issues

Comment (p0mvn) [reply cmd/osmosisd/cmd/root.go:519]: [33a0a44](https://github.com/osmosis-labs/osmosis/pull/7368/commits/33a0a443f43e773716baf60908cd3458401c4aef)

**Comment by hleb-albau** [comment]
why not make it just 1 osmo and leave this spam ahead?

**Comment by p0mvn** [comment]
> why not make it just 1 osmo and leave this spam ahead?

Generally, I agree with increasing it even higher. I was worried about making too much of a rapid change for all nodes at once. This is already a 20x increase

**Pull request merged**